motionplan/motionPlanner.go

```diff
@@ -54,10 +54,57 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
```

Comment:
if the request is `nil` this will panic, which is possible as PlanMotion takes a pointer: https://github.com/viamrobotics/rdk/pull/3009/files#diff-aa6bcc687ca0f9ce94ee44319a1e46c90ee0c570db7a27250f4ae1a8de2fd736R104

we should check for nil & reutrn an error if so